src/AppInstallerCLICore/Workflows/ManifestComparator.cpp

```diff
@@ -46,17 +46,19 @@ namespace AppInstaller::CLI::Workflow
 
         struct MachineArchitectureComparator : public details::ComparisonField
         {
-            MachineArchitectureComparator() : details::ComparisonField("Machine Architecture") {}
+            MachineArchitectureComparator(Utility::Architecture preference) :
+        	details::ComparisonField("Machine Architecture"), m_preference(preference) {}
 
-            MachineArchitectureComparator(std::vector<Utility::Architecture> allowedArchitectures) :
-                details::ComparisonField("Machine Architecture"), m_allowedArchitectures(std::move(allowedArchitectures))
+            MachineArchitectureComparator(std::vector<Utility::Architecture> allowedArchitectures, Utility::Architecture preference) :

```

Comment:
This shouldn't have needed to change to fully support preferences and requirements.  An entry of `Utility::Architecture::Unknown` in `Execution::Data::AllowedArchitectures` indicates "and all the rest"; aka that the previous architectures are preferences.  Thus the only difference between the preferences and requirements settings are whether to append `Unknown`.